webapp/src/ts/modals/navigation-confirm/navigation-confirm.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+import { MmModalAbstract } from '../mm-modal/mm-modal';
+
+@Component({
+  selector: 'navigation-confirm-modal',
+  templateUrl: './navigation-confirm.component.html'
+})
+export class NavigationConfirmComponent extends MmModalAbstract {
+  constructor(
+    public bsModalRef: BsModalRef,
+  ) {
+    super();
+  }
+
+  callback; // Automatically assigned by BsModalRef
```

Comment:
This looks pretty unusual. Does the functionality not work without doing this? I'm having a hard time following how this works; do you have a documentation link I can read to better understand?

**update:** Is that what's going on here? `webapp/src/ts/effects/global.effects.ts:73` If that's the case, what happens if the `callback` property is omitted? Will things blow up down in the submit method below?

Reply:
I'm not really happy with this.
The modal service from AngularJS returned a promise that worked kinda like a confirm, resolved on ok, rejected on cancel. All code is structured assuming that we can then a modal. But we can't now.

I was going to spend some time to find a proper solution for this. @mrsarm also has a modal that's supposed to trigger some other action AFTER it's resolved.

Reply:
I've changed this completely.